jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryEndpointCaller.java

```diff
@@ -265,70 +252,73 @@ private T call(URL url, Function<URL, Connection> connectionFactory)
+            || responseException.getStatusCode() == HttpStatusCodes.STATUS_CODE_NOT_FOUND
+            || responseException.getStatusCode()
                 == HttpStatusCodes.STATUS_CODE_METHOD_NOT_ALLOWED) {
           // The name or reference was invalid.
-          throw newRegistryErrorException(httpResponseException);
+          throw newRegistryErrorException(responseException);
 
-        } else if (httpResponseException.getStatusCode() == HttpStatusCodes.STATUS_CODE_FORBIDDEN) {
-          throw new RegistryUnauthorizedException(
-              registryEndpointRequestProperties.getServerUrl(),
-              registryEndpointRequestProperties.getImageName(),
-              httpResponseException);
+        } else if (responseException.getStatusCode() == HttpStatusCodes.STATUS_CODE_FORBIDDEN) {
+          throw new RegistryUnauthorizedException(serverUrl, imageName, responseException);
 
-        } else if (httpResponseException.getStatusCode()
-            == HttpStatusCodes.STATUS_CODE_UNAUTHORIZED) {
+        } else if (responseException.getStatusCode() == HttpStatusCodes.STATUS_CODE_UNAUTHORIZED) {
           if (sendCredentials) {
             // Credentials are either missing or wrong.
-            throw new RegistryUnauthorizedException(
-                registryEndpointRequestProperties.getServerUrl(),
-                registryEndpointRequestProperties.getImageName(),
-                httpResponseException);
+            throw new RegistryUnauthorizedException(serverUrl, imageName, responseException);
           } else {
-            throw new RegistryCredentialsNotSentException(
-                registryEndpointRequestProperties.getServerUrl(),
-                registryEndpointRequestProperties.getImageName());
+            throw new RegistryCredentialsNotSentException(serverUrl, imageName);
           }
 
-        } else if (httpResponseException.getStatusCode()
+        } else if (responseException.getStatusCode()
                 == HttpStatusCodes.STATUS_CODE_TEMPORARY_REDIRECT
-            || httpResponseException.getStatusCode()
-                == HttpStatusCodes.STATUS_CODE_MOVED_PERMANENTLY
-            || httpResponseException.getStatusCode() == STATUS_CODE_PERMANENT_REDIRECT) {
+            || responseException.getStatusCode() == HttpStatusCodes.STATUS_CODE_MOVED_PERMANENTLY
+            || responseException.getStatusCode() == STATUS_CODE_PERMANENT_REDIRECT) {
           // 'Location' header can be relative or absolute.
-          URL redirectLocation = new URL(url, httpResponseException.getHeaders().getLocation());
+          URL redirectLocation = new URL(url, responseException.getHeaders().getLocation());
           return callWithAllowInsecureRegistryHandling(redirectLocation);
 
         } else {
           // Unknown
-          throw httpResponseException;
+          throw responseException;
         }
       }
+
+    } catch (SSLException ex) {
+      throw ex;
```

Comment:
just wondering what this is about?

Reply:
So originally this `IOException` catch block was in this method. Then I moved it into another method in [this PR](https://github.com/GoogleContainerTools/jib/pull/1950/files#diff-1aa0b60da6c25b066e47abc6ebe10509) while adding logging for every `IOException`. The main reason I moved it back then was that almost all the time I shouldn't be logging errors for `SSLException` (which extends `IOException`), because `SSLException` as of now usually is expected for TLS failover. And since we are going to take out the TLS failover from this class, I now want to move this block back to its original place mostly for syntactically better-looking code. But to do so, I need to add this weird catch block to not log errors for `SSLException`.

But I just realized I should still need to log broken pipe in the `SSLException` case too after checking the [history](https://github.com/GoogleContainerTools/jib/issues/1316#issuecomment-478100424) behind it. I'll push up a small change.

Reply:
But once the TLS failover is decoupled from this class, I should be able to log errors for all `IOException` including `SSLException`, since `SSLException` in the new world will conclusively mean that the `SSLException` is final and you can no longer proceed.